src/Core/Navigation/INavigationService.cs

```diff
@@ -20,63 +20,68 @@ public interface INavigationService
         /// <summary>
         /// Removes an existing mapping between a ViewModel and a View
         /// </summary>
-        /// <typeparam name="TViewModelInterface"></typeparam>
-        void Unregister<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        /// <typeparam name="TViewModel"></typeparam>
+        void Unregister<TViewModel>()
+            where TViewModel : class, IViewModel;
 
 
+        /// <summary>
+        /// Removes all existing mappings between ViewModel and View types
+        /// </summary>
+        void ResetRegistrations();
+
 
         /// <summary>
         /// Sets the application's root view using the ViewModel interface and the registered View type
         /// </summary>
-        /// <typeparam name="TViewModelInterface"></typeparam>
+        /// <typeparam name="TViewModel"></typeparam>
         /// <param name="dataContext">A ViewModel instance of the given interface type</param>
-        Task SetRootAsync<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        Task SetRootAsync<TViewModel>()
+            where TViewModel : class, IViewModel;
 
 
         /// <summary>
         /// Creates a new instance of the ViewModel interface type, looks up the associated view and displays it
         /// </summary>
-        Task ShowAsync<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        Task ShowAsync<TViewModel>()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to have parameters matching all the type parameters.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)